test/test_autograd.py

```diff
@@ -5336,6 +5336,42 @@ def assert_only_first_requires_grad(res):
                                          return_counts=return_counts)
                     assert_only_first_requires_grad(res)
 
+    def test_custom_function_cycle(self):
+        class MyFn(Function):
+            @staticmethod
+            def forward(ctx, x):
+                x = x.clone()
+                # BAD!!! Don't do this ever!!
+                ctx.x = x
+                return x
+
+            @staticmethod
+            def backward(ctx, gO):
+                self.assertEqual(ctx.x, 3.14)
+                return gO * ctx.x
+
+        def get_refs(with_backward):
+            a = torch.tensor(3.14, requires_grad=True)
+
+            out = MyFn.apply(a)
+
+            if with_backward:
+                out.sum().backward()
+                self.assertEqual(a.grad, a)
+
+            return torch._C._WeakTensorRef(out)
+
+        ref = get_refs(False)
+        self.assertFalse(ref.expired())
```

Comment:
GC collection can trigger at any point so this test can be flaky. You should explicitly disable GC, in the same way I do in my pending patch: https://github.com/pytorch/pytorch/pull/56017/files#diff-8aa1a200ec63d23db422aa31b6dca1e6cb372887c43b064ef435210b1b0dec0aR8218-R8226

Reply:
Yes, I didn't knew we could do that. I'll update this!